cmd/lotus-miner/actor.go

```diff
@@ -206,10 +211,307 @@ var actorSetPeeridCmd = &cli.Command{
 	},
 }
 
+var actorWithdrawMsigProposeCmd = &cli.Command{
+	Name:      "propose",
+	Usage:     "Propose a multisig transaction",
+	ArgsUsage: "[multisigAddress destinationAddress value <methodId methodParams> (optional)]",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "encoding",
+			Value: "hex",
+			Usage: "specify params encoding to parse (base64, hex)",
+		},
+		&cli.StringFlag{
+			Name:  "from",
+			Usage: "account to send the propose message from",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		if cctx.Args().Len() < 3 {
+			return lcli.ShowHelp(cctx, fmt.Errorf("must pass at least multisig address, destination, and value"))
+		}
+
+		if cctx.Args().Len() > 3 && cctx.Args().Len() != 5 {
+			return lcli.ShowHelp(cctx, fmt.Errorf("must either pass three or five arguments"))
+		}
+
+		srv, err := lcli.GetFullNodeServices(cctx)
+		if err != nil {
+			return err
+		}
+		defer srv.Close() //nolint:errcheck
+
+		api := srv.FullNodeAPI()
+		ctx := lcli.ReqContext(cctx)
+
+		msig, err := address.NewFromString(cctx.Args().Get(0))
+		if err != nil {
+			return err
+		}
+
+		dest, err := address.NewFromString(cctx.Args().Get(1))
+		if err != nil {
+			return err
+		}
+
+		value, err := types.ParseFIL(cctx.Args().Get(2))
+		if err != nil {
+			return err
+		}
+
+		available, err := api.StateMinerAvailableBalance(ctx, msig, types.EmptyTSK)
+		if err != nil {
+			return err
+		}
+
+		var method uint64
+		var params []byte
+		if cctx.Args().Len() == 5 {
+			m, err := strconv.ParseUint(cctx.Args().Get(3), 10, 64)
+			if err != nil {
+				return err
+			}
+			method = m
+
+			// get amount
+			amount := available
+			if cctx.Args().Present() {
+				f, err := types.ParseFIL(cctx.Args().Get(4))
+				if err != nil {
+					return xerrors.Errorf("parsing 'amount' argument: %w", err)
+				}
+
+				amount = abi.TokenAmount(f)
+
+				if amount.GreaterThan(available) {
+					return xerrors.Errorf("can't withdraw more funds than available; requested: %s; available: %s", types.FIL(amount), types.FIL(available))
+				}
+			}
+
+			params, err = actors.SerializeParams(&miner.WithdrawBalanceParams{
+				AmountRequested: amount, // Default to attempting to withdraw all the extra funds in the miner actor
+			})
+			if err != nil {
+				return err
+			}
+
+		}
+
+		var from address.Address
+		if cctx.IsSet("from") {
+			f, err := address.NewFromString(cctx.String("from"))
```

Comment:
just assign `from` here, instead of assigning `f` here and then assigning `from` to `f`